src/pricecypher/datasets.py

```diff
@@ -0,0 +1,396 @@
+        self._check_types()
+
+    def _check_types(self):
+        for v in self._list:
+            if not isinstance(v, ScopeValue):
+                raise TypeError(v)
+
+    def __repr__(self):
+        return "<{0} {1}>".format(self.__class__.__name__, self._list)
+
+    def __len__(self):
+        """List length"""
+        return len(self._list)
+
+    def __getitem__(self, ii):
+        """Get a list item"""
+        return self._list[ii]
+
+    def __str__(self):
+        return str(self._list)
+
+    def where_in(self, values):
+        """
+        Filter collection on the given values.
+
+        :param list or float or str values: Value or values to filter the collection on.
+        :return: Collection of filtered scope values.
+        :rtype: ScopeValueCollection
+        """
+        # Turn values into a list if it is not a list already.
+        if type(values) is not list:
+            values = [values]
+
+        # Make sure all values are strings.
+        values = list(map(str, values))
+        # Filter and create new collection
+        scope_values = [sv for sv in self._list if sv.value in values]
+        return ScopeValueCollection(scope_values)
+
+    def pluck(self, prop):
+        """
+        Get a list with the values of the given prop.
+
+        :param str prop: Property to pluck from the scope values.
+        :return:
+        """
+        return [getattr(v, prop) for v in self._list]
+
+
+class Datasets(object):
+    """
+    Datasets API. Exposes all available operations on datasets, like fetching the available datasets, listing
+    the scopes within a dataset, retrieving transactions within a dataset, etc.
+
+    :param str bearer_token: Bearer token for PriceCypher (logical) API. Needs 'read:datasets' scope.
+    :param str users_base: (optional) Base URL for PriceCypher user tool API. Used to find base URL of dataset service,
+        when no dss_base is provided.
+        (defaults to https://users.pricecypher.com)
+    :param str dss_base: (optional) Base URL for PriceCypher dataset service API.
+        (defaults to dss_url property as returned for the dataset by the PriceCypher user tool API)
+    :param RestClientOptions rest_options: (optional) Set any additional options for the REST client, e.g. rate-limit.
+        (defaults to None)
+    """
+
+    def __init__(self, bearer_token, users_base='https://users.pricecypher.com', dss_base=None, rest_options=None):
+        self._bearer = bearer_token
+        self._users_base = users_base
+        self._dss_base = dss_base
+        self._rest_options = rest_options
+        self._client = RestClient(jwt=bearer_token, options=rest_options)
+        self._all_meta = None
+
+    def _ds_base(self, dataset_id):
+        """
+        Get dataset service url base for the given dataset ID.
+        Will be fetched from dataset META if no dss_base present.
+
+        :param int dataset_id: Dataset ID to get base URL for.
```

Comment:
Good catch, updated. It was probably working fine during testing since you only queried datasets which have a dss_url specified in the user tool (which should be all datasets that are contained in some dss instance).